torch/csrc/jit/codegen/cuda/scheduler/mma_utils.cpp

```diff
@@ -302,6 +389,101 @@ void scheduleVoltaB(TensorView* tv, MmaOptions options) {
   tv->axis(-2)->parallelize(ParallelType::TIDx);
 }
 
+void scheduleLdMatrix(TensorView* tv, MmaOptions options) {
+  // Check if tv should use ldmatrix layout and
+  //   if tv is immediate output of ldmatrix
+  bool is_immediate_output = checkLdMatrixTv(tv);
+
+  // Decode transposition requirement for turing mma
+  bool transposed = options.operand == MmaOptions::Operand::A
+      ? !isOperandTransposed(options)
+      : isOperandTransposed(options);
+  // Check mma option is supported
+  TORCH_CHECK(
+      options.macro == MmaOptions::MacroType::Ampere_16_8_16,
+      "scheduleLdMatrix: unknown macro for ldmatrix");
+
+  if (options.operand == MmaOptions::Operand::A) {
+    TORCH_INTERNAL_ASSERT(tv->nDims() >= 2);
+    // validation:
+    TORCH_INTERNAL_ASSERT(canValidateIsInnerDim(
```

Comment:
Nit: Asserts should have something printed unless it should be impossible to reach.

Reply:
Added error message in these checks. Thanks.